src/app/profile-page/profile-page-researcher-form/profile-page-researcher-form.component.ts

```diff
@@ -0,0 +1,190 @@
+   *
+   * @param researcherProfile the current researcher profile
+   */
+  viewProfile(researcherProfile: ResearcherProfile): void {
+    if (this.researcherProfileItemId != null) {
+      this.router.navigate(['items', this.researcherProfileItemId]);
+    }
+  }
+
+  /**
+   * Delete the given researcher profile.
+   *
+   * @param researcherProfile the profile to delete
+   */
+  deleteProfile(researcherProfile: ResearcherProfile): void {
+    this.processingDelete$.next(true);
+    this.researcherProfileService.delete(researcherProfile)
+      .subscribe((deleted) => {
+        if (deleted) {
+          this.researcherProfile$.next(null);
+          this.researcherProfileItemId = null;
+        }
+        this.processingDelete$.next(false);
+      });
+  }
+
+  /**
+   * Toggle the visibility of the given researcher profile.
+   *
+   * @param researcherProfile the profile to update
+   */
+  toggleProfileVisibility(researcherProfile: ResearcherProfile): void {
+    this.researcherProfileService.setVisibility(researcherProfile, !researcherProfile.visible)
+      .subscribe((rd: RemoteData<ResearcherProfile>) => {
+        if (rd.hasSucceeded) {
+          this.researcherProfile$.next(rd.payload);
+        } else {
+          this.notificationService.error(null, this.translationService.get('researcher.profile.change-visibility.fail'));
+        }
+      });
+  }
+
+  /**
+   * Return a boolean representing if a delete operation is pending.
+   *
+   * @return {Observable<boolean>}
+   */
+  isProcessingDelete(): Observable<boolean> {
+    return this.processingDelete$.asObservable();
+  }
+
+  /**
+   * Return a boolean representing if a create operation is pending.
+   *
+   * @return {Observable<boolean>}
+   */
+  isProcessingCreate(): Observable<boolean> {
+    return this.processingCreate$.asObservable();
+  }
+
+  /**
+   * Create a new profile related to the current user from scratch.
+   */
+  createProfileFromScratch() {
+    this.processingCreate$.next(true);
+    this.researcherProfileService.create().pipe(
+      getFirstCompletedRemoteData()
+    ).subscribe((remoteData) => {
+      this.processingCreate$.next(false);
+      if (remoteData.isSuccess) {
+        this.initResearchProfile();
+        this.notificationService.success(null, this.translationService.get('researcher.profile.create.success'));
+      } else {
+        this.notificationService.error(null, this.translationService.get('researcher.profile.create.fail'));
+      }
+    });
+  }
+
+  /**
+   * Initializes the researcherProfile and researcherProfileItemId attributes using the profile of the current user.
+   */
+  private initResearchProfile(): void {
+    this.researcherProfileService.findById(this.user.id, false).pipe(
+      getFirstSucceededRemoteDataPayload(),
+      tap((researcherProfile) => this.researcherProfile$.next(researcherProfile)),
+      mergeMap((researcherProfile) => this.researcherProfileService.findRelatedItemId(researcherProfile)),
```

Comment:
perhaps add `followLink('item')` to the `findById` call above? Then you wouldn't have to do this mergeMap, and the rest api can embed the item in the original call.

Reply:
added followLink and changed the findRelatedItemId in order to use embedded item if available